### DISCUSSION TITLE
yaml-cpp in custom material

### DISCUSSION THREAD

**Discussion by gsgall** [idaholab/moose#25865] [Q&A General]
I am trying to use the yaml-cpp library in a custom material property. I believe wasp depends on this library and I see a dynamic library file in `mambaforge3/envs/moose/lib`. Specifically `libyaml.dylib`

I am able to compile my custom material with  `#include yaml.h` but unable to use the library as suggested on the tutorial for the project . 

This is the only code I have in my material initializer at the moment

```// check if the file exists and we can read it
  std::string file_name = getParam<FileName>("reaction_file");
  MooseUtils::checkFileReadable(file_name);

  YAML::Node network = YAML::LoadFile(file_name);
```

Is there something else I need to include to be able to use this library within a MOOSE material? Or does this library link to another yaml reader project?

**Comment by GiudGiud** [comment]
@brandonlangley @loganharbour if you have experience with YAML

**Reply by brandonlangley** [reply]
@GiudGiud -

I'm not familiar with the yaml-cpp library, and I don't think this dependency is coming from WASP.

WASP has never done any type of YAML processing, so unfortunately I can't help here.

**Comment by loganharbour** [comment]
We don't have any framework YAML readers or writers at the moment.

We instead use JSON quite extensively, primarily through https://github.com/nlohmann/json.

**Reply by milljm** [reply]
Pining @MengnanLi91, due to the conversation we had about YAML and JSON (donno, maybe the info here may be of some instrest?)
